src/OpalCompiler-Core/RBMethodNode.extension.st

```diff
@@ -167,7 +167,8 @@ RBMethodNode >> rewriteTempsForContext: aContext [
 
 { #category : #'*OpalCompiler-Core' }
 RBMethodNode >> sourceNodeForPC: anInteger [
-	^(self ir instructionForPC: anInteger) sourceNode
+	bcToASTCache ifNil: [ bcToASTCache := OCBytecodeToASTCache generateForMethodNode: self].
```

Comment:
If we add a lazy accessor to bcToASTCache, we could just have `^ bcToASTCache nodeForPC: anInteger`.

What do you think?

Reply:
Yes!